gui

```diff
@@ -502,49 +458,110 @@ if [[ "$guimode" == yad* ]];then
   trap "rm -f $pipe" EXIT #remove this named pipe on exit
   echo pipe is $pipe
   
-  while true;do #main operating loop for yad window
-    #in background process, send initial app list to yad; first echo line waits until yad is accepting input, then use small delay to prevent items from being missed
-    (echo -e '\f' > $pipe
-    sleep 0.5
-    "${DIRECTORY}/preload" yad '' >> $pipe) &
-    
-    #retrieve a random line from the announcements file for this session
-    motd="$(shuf -n 1 "${DIRECTORY}/data/announcements")"
+  #in background process, send initial app list to yad; first echo line waits until yad is accepting input, then use small delay to prevent items from being missed
+  (echo -e '\f' > $pipe
+  sleep 0.5
+  "${DIRECTORY}/preload" yad '' >> $pipe) &
+  
+  #retrieve a random line from the announcements file for this session
+  motd="$(shuf -n 1 "${DIRECTORY}/data/announcements")"
+  
+  return_sysname() { #Function that is run in yad gui to echo the selected line to the listening while loop.
+    echo "YAD_XID is $YAD_XID"
+    echo "$3"
+  }
+  
+  refresh_list() { #Refresh the current list of apps in the event of a change
+    if [ ! -z "$pipe" ];then
+      echo -e "\f" > "$pipe"
+      "${DIRECTORY}/preload" yad "$prefix" > "$pipe" 2>/dev/null
+    fi
+  }
+  
+  kill_details() { #close the details window in 1 second. Uses the $detailspid variable. To close it immediately add an argument of 0
+    (
+    if [ "$1" != 0 ];then
+      sleep 1.5
+    fi
+    pkill -P "$detailspid" 2>/dev/null
```

Comment:
this kills child PIDs of the parent process (the details window)
this (I think) will kill any running or backgrounded `terminal_manage` calls spawned from that detailspid

Reply:
I had not considered this possibility, but it does not seem to kill the terminal in practice. Not sure why not.

Reply:
I think in practice this is impossible to hit because in order to kill the detailspid that has the manage child process, you would need to click uninstall/install and another app name in the list at the exact same time. Since the partent process (the details window) pid disappears once the yad details window closes on clicking the install/uninstall button, the kill_details pkill does nothing since that pid does not exist (even though it has child PIDs).